### PR TITLE
Allow add multiple events on campus wizzard

### LINKED ISSUE #154: new action: add events
See mockup 'Campus Form - Important Dates'.
The mockup is not correct: it should only have one button 'add another event'.
When clicked, it should show three new inputs:
- event name (text)
- event date (date
- http link to event (text)

### PR DESCRIPTION
**Pull request by Christiaanvdm**
This closes #154 
NewGIF

### CODE REVIEW

File: django_project/ford3/templates/campus_event.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<script type="text/template" id="add-campus-event">
+    <input type="hidden" name="campus_form_wizard-current_step" value="2" id="id_campus_form_wizard-current_step">
+
+    <input type="hidden" name="csrfmiddlewaretoken"
```
Comment (pierrealixt) [inline django_project/ford3/templates/campus_event.html]: this is generated by django.

File: django_project/ford3/templates/campus_form.html [hunk 2]
```diff
@@ -46,7 +46,9 @@ <h4 class="list-group-item-heading">
                         {% else %}
                             {% crispy wizard.form wizard.form.helper %}
                         {% endif %}
-
+                        {% if wizard.steps.step1 == 3 %}
+	                        Ok so where will this go?
```
Comment (pierrealixt) [inline django_project/ford3/templates/campus_form.html]: maybe you want to remove this?

File: django_project/ford3/views/campus_wizard.py [hunk 3]
```diff
@@ -73,11 +77,58 @@ def done(self, form_list, **kwargs):
                 self.campus.save_form_data(form.cleaned_data)
             elif i == steps['DATES']:
                 print(form.cleaned_data)
```
Comment (pierrealixt) [inline django_project/ford3/views/campus_wizard.py]: remove the print.

File: django_project/ford3/views/campus_wizard.py [hunk 4]
```diff
@@ -72,12 +75,57 @@ def done(self, form_list, **kwargs):
...
+
+        if step_data['campus_form_wizard-current_step'][0] == '2':
+            new_name = step_data['2-name']
+            new_date_start = step_data['2-date_start']
+            new_date_end = step_data['2-date_end']
+            new_http_link = step_data['2-http_link']
+
+
+
+            for i in range(0, len(new_name)):
```
Comment (pierrealixt) [inline django_project/ford3/views/campus_wizard.py]: `range(0, len(new_name))` ?

File: django_project/ford3/views/campus_wizard.py [hunk 5]
```diff
@@ -72,12 +75,57 @@ def done(self, form_list, **kwargs):
...
+                new_campus_event.date_end = new_date_end_formatted
+                new_campus_event.http_link = new_http_link[i]
+                self.new_campus_events.append(new_campus_event)
+
+
+    def render(self, form=None, **kwargs):
+        form = form or self.get_form()
+        context = self.get_context_data(form=form, **kwargs)
+        current_step = context['view'].storage.current_step
+        if current_step == '3' or current_step == '2':
```
Comment (pierrealixt) [inline django_project/ford3/views/campus_wizard.py]: why `current_step == '3' or current_step == '2'`?

File: django_project/ford3/tests/models/test_campus.py [hunk 6]
```diff
@@ -43,33 +42,23 @@ def test_save_location_form_data(self):
         pass
 
     def test_save_events_form_data(self):
-        form_data = {
-            'event_type': 'Open day',
-            'event_date': datetime.date(2019, 3, 30),
-            'event_http_link': 'http://event42.com'
-        }
+
+        campus_events = [ModelFactories.get_campus_event_test_object(36)]
```
Comment (pierrealixt) [inline django_project/ford3/tests/models/test_campus.py]: 36?

File: django_project/ford3/views/campus_wizard.py [hunk 7]
```diff
@@ -72,12 +75,60 @@ def done(self, form_list, **kwargs):
...
             elif i == steps['DATES']:
-                print(form.cleaned_data)
-                self.campus.save_events(form.cleaned_data)
+                self.campus.save_events(self.new_campus_events)
             elif i == steps['QUALIFICATION_TITLES']:
                 self.campus.save_qualifications(form.cleaned_data)
                 self.campus.delete_qualifications(form.cleaned_data)
             i += 1
+        return redirect(
+            '/ford3/providers/{provider_id}/campus/{campus_id}'.format(
```
Comment (pierrealixt) [inline django_project/ford3/views/campus_wizard.py]: use `reverse`

**Comment by Christiaanvdm** [comment]
I don't know why the selenium tests are not executing the javascript in travis? Would it not use the StaticLiveServerTestCase as specified in utils.py?

**Review by pierrealixt — APPROVED** [review]
LGTM

**Pull request merged**